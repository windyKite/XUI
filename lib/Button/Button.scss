.Xui-button {
  display: inline-block;
  margin-bottom: 0;
  font-weight: 400;
  text-align: center;
  vertical-align: middle;
  cursor: pointer;
  outline: none;
  white-space: nowrap;
  user-select: none;
  padding: 8px 12px;
  font-size: 14px;
  line-height: 1.42857143;
  border-radius: 6px;
  background-image: none;
  border: 1px solid #ccc;

  & > svg{
    vertical-align: middle;
    padding-right: 6px;
  }

  &.disabled {
    cursor: not-allowed !important;
    opacity: .4;
    color: #c5c6c7;
    background-color: #f7f7fa!important;
  }

  &.default {
    color: #575757;
    background: #f7f7fa;

    &:hover:not(.disabled) {
      background-color: #e5e5ea;
    }
  }

  &.primary {
    color: #fff;
    background-color: #3498ff;

    &:hover:not(.disabled) {
      background-color: #2589f5;
    }
  }

  &.subtle {
    color: #8e8e93;
    background-color: transparent;

    &:hover:not(.disabled) {
      background-color: #f7f7fa;
    }
  }

  &.ghost {
    color: #1675e0;
    background-color: transparent;
    border-color: #1675e0;

    &:hover:not(.disabled) {
      color: #0a5dc2;
      border-color: #0a5dc2;
    }
  }

  &.link {
    color: #1675e0;
    border: none;

    &:hover {
      text-decoration: underline;
    }
  }
  &.large{
    padding: 10px 16px;
  }
  &.small{
    padding: 6px 8px;
  }
}

.ripple:not(.link.disabled) {
  position: relative;
  overflow: hidden;
  transform: translate3d(0, 0, 0);

  &::after {
    content: "";
    display: block;
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    pointer-events: none;
    background-image: radial-gradient(circle, #000 10%, transparent 10.01%);
    background-repeat: no-repeat;
    background-position: 50%;
    transform: scale(10, 10);
    opacity: 0;
    transition: transform .5s, opacity 1s;
  }

  &:active:after {
    transform: scale(0, 0);
    opacity: .2;
    transition: 0s;
  }
}