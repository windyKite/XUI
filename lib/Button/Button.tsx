import React from "react";
import Icon from '../Icon/Icon'
import classNames from "classnames";
import "./Button.scss"

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  icon?: string;
  appearance?: "default" | "primary" | "subtle" | "ghost";
  disabled?: boolean;
  href?: string;
  size?: "large" | "small";
}

const Button: React.FunctionComponent<ButtonProps> = ({
  className,
  icon,
  appearance = "default",
  color,
  disabled,
  href,
  size,
  children,
  ...restProps
}) => {
  const classes = classNames("Xui-button", "ripple", className,appearance, color, size, { disabled })

  if(icon){
    return (
      <button className={classes} {...restProps}>
        <Icon name={icon}/>
        {children}
      </button>
    )
  }else{
    return (
      <button className={classes} {...restProps}>
        {children}
      </button>
    )
  }
}

export default Button