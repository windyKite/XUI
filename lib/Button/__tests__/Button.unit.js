import * as renderer from "react-test-renderer";
import React from "react";
import Button from "../Button"
import { mount } from "enzyme";

describe("Button", () => {
  it("Button", () => {
    const json = renderer.create(<Button appearance="ghost"/>).toJSON();
    expect(json).toMatchSnapshot();
  });
  it("Button click", () => {
    const fn = jest.fn();
    const fn2 = jest.fn()
    const component = mount(<Button onClick={fn} />);
    component.find("svg").simulate("click");
    expect(fn).toBeCalled();
  })
});
