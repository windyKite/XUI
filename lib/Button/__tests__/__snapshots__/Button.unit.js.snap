// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Button 1`] = `
<button
  className="Xui-button ripple ghost"
/>
`;
